test/chplcheck/MethodsAfterFields/MethodsAfterFields.prediff

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+while read expectedLine; do
+    echo grep "$expectedLine" MethodsAfterFields.message | echo $? >> MethodsAfterFields.comp.out.tmp
```

Comment:
Separately, prediff files get the name of the `.comp.out.tmp` file passed as their `$2` argument, and the test name (e.g. `MethodsAfterFields`) as `$1`. 

https://github.com/chapel-lang/chapel/blob/48128ec6673211eb4f159bdcc2e95eb6fc9fe098/util/test/sub_test.py#L1786-L1788

Putting this together with my other suggestions, you might be able to get away with:

```bash
if diff $1.message $1.expected; then
    echo "Linter output matched!" >> $2
else
    echo "Linter output did not match!" >> $2
fi
```

The advantage here is that this prediff would work for other tests in this style.

Reply:
The $1 and $2 parts sound good to me, but the message outputted by the linter uses the absolute path of the file with the problems, so I'm not sure that the `diff` plan would work across machines without some scheme that expands CHPL_HOME within the expected output.